src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -235,11 +236,51 @@ private void ProcessEndRequest(object eventSender)
 			var httpApp = (HttpApplication)eventSender;
 			var httpCtx = httpApp.Context;
 			var httpResponse = httpCtx.Response;
+			var transaction = _currentTransaction;
 
-			if (_currentTransaction == null) return;
+			if (transaction == null) return;
 
 			SendErrorEventIfPresent(httpCtx);
 
+			if (!transaction.HasCustomName)
+			{
+				//fixup Transaction.Name - e.g. /user/profile/1 -> /user/profile/{id}
+				var values = httpApp.Request.RequestContext?.RouteData?.Values;
+
+				if (values?.Count > 0 && httpResponse.StatusCode != StatusCodes.Status404NotFound)
```

Comment:
One (I think) fairly common case which is not handled is when a REST API returns 404.

[Here](https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-http.md) is the spec.

It says: 
>  This would normally also apply to requests to unknown endpoints, e.g. the transaction for the request GET /this/path/does/not/exist would be named GET unknown route, whereas the transaction for the request GET /users/123 would still be named GET /users/{id}

If we have `/user` endpoint which returns 404 for non-existent users then we'd skip the if block and keep the existing transaction name based on the url (so e.g. `GET /user/123` if `123` doesn't exist). And this would cause transaction name explosion.

Reply:
Good catch! We can differentiate here based on the presence of a 404 HttpException on the HttpContext coming from System.Web.Mvc